src/Utility/ExprFunctions.C

```diff
@@ -205,26 +226,26 @@ EvalFunction::EvalFunction (const char* function) : gradient{}, dgradient{}
 
 EvalFunction::~EvalFunction ()
 {
-  cleanup();
+  this->cleanup();
 }
 
 
-void EvalFunction::cleanup()
+void EvalFunction::cleanup ()
 {
-  for (auto& it : expr)
+  for (ExprEval::Expression* it : expr)
     delete it;
-  for (auto& it : f)
+  for (ExprEval::FunctionList* it : f)
     delete it;
-  for (auto& it : v)
+  for (ExprEval::ValueList* it : v)
     delete it;
-  for (auto& it : gradient)
+  for (EvalFunction* it : gradient)
```

Comment:
clear these as well i guess.

Reply:
not sure about that. gradient (and dgradient) are std::array's and not vectors. Then clear is not defined. But do we need to nullify the array members (pointers) or is that automatically done through the delete operator.

Reply:
delete does not nullify, std::fill or for() then.

Reply:
std::array has a fill member, so used that instead.